### DISCUSSION TITLE
Rerun failed test issue when in parallel running mode

### DISCUSSION THREAD

**Discussion by windyliu1118** [getgauge/gauge#1914] [General]
I have set my automation test running with 2 process (2 CPU) using command `gauge run --max-retries-count=2 --env staging --parallel -n=2 specs`, and got 4 failed tests.
<img width="1079" alt="Screen Shot 2021-02-18 at 12 16 57 PM" src="https://user-images.githubusercontent.com/7007401/108304352-d2e98900-71e2-11eb-90e8-c028d0dfea98.png">

And then there is one command `gauge run --failed` to run the failed tests, and then I got the result that 7 tests executed.
<img width="983" alt="Screen Shot 2021-02-18 at 12 16 19 PM" src="https://user-images.githubusercontent.com/7007401/108304399-edbbfd80-71e2-11eb-8644-d5cb8f1464d4.png">

This is really weird, why 4 tests failed, but the rerun failed actually executed 7 tests?

**Comment by windyliu1118** [comment]
Anyone can take a look?

**Reply by sriv** [reply]
what does your `.gauge/failures.json` contain? This file is used to record the failures from previous run

**Reply by windyliu1118** [reply]
OK, I will rerun the tests in parallel and check the `.gauge/failures.json` file.
